submissions/fluidic-power/fluidic-power.md

```diff
@@ -2,57 +2,57 @@
+This showed something that I did not really expect. The fluid system (`FluidSystem::update`) was only part of the top 10 worst timed functions and the Electric Network was _still_ taking more processing time! It turns out the main problem is `FlowStatistics<ID<...>...>::onFlow` which is called mostly by the electric network. These [flow statistics](https://lua-api.factorio.com/latest/LuaFlowStatistics.htm) are simply used to collect statistics so that you can view it on in the graphs, eg. for power production. This means that that the game is not only slowing down because of all the new fluid calculations, but also because of all the new graphs to draw!
 
-It's very likely that Electric Network statistics collection could take a significant amount of processing time in Fluidic Power. The Factorio engine is built to only have a handful of electric networks, normally even only one. The game needs to collect and update information for every power network you have, which will be only a few of datasets. So instead of having a handful of networks Fluidic Power is designed so that _each and every_ power pole is a single electric network. The Factorio engine will then attempt to collect statistics for each of these electric networks - which in my benchmark save is about 3000! That's at least 100 times more networks to collect statistics on than Factorio is designed for (or usually). This has been a known limitation for some other mods, but Fluidic Power takes this to a new extreme. However, we still don't know the impact of these calculations _relative_ to the fluid calculations. Does the collecting of statistics really take that much more time than fluid calculations?
+It's very likely that `Electric Network` statistics collection could take a significant amount of processing time in Fluidic Power. The Factorio engine is designed to only have a handful of electric networks, often only a single one even. The game needs to collect and update information for every power network you have, which will normally only be a few distinct datasets. So instead of having a handful of networks, Fluidic Power is designed so that _each and every_ power pole is a single electric network. The Factorio engine will then attempt to collect statistics for each of these electric networks - which in my benchmark save is about 3000! That's at least 100 times more networks to collect statistics on than Factorio is designed for. This has been a known limitation for some other mods, but Fluidic Power takes this to a new extreme. However, we still don't know the impact of these calculations _relative_ to the fluid calculations. Does the collecting of statistics really take that much more time than the fluid calculations themselves?
```

Comment:
```suggestion
It's very likely that the collection of `Electric Network` statistics could take a significant amount of processing time with Fluidic Power. The Factorio engine is designed to only have a handful of electric networks, or even just a single one. The game needs to collect and update information for every power network you have, which will normally only be a few distinct datasets. So instead of having a handful of networks, Fluidic Power is designed so that _each and every_ power pole is a single electric network. The Factorio engine will then attempt to collect statistics for each of these electric networks - which is about 3000 electric networks in my benchmark save! That's at least 100 times more networks to collect statistics on than Factorio is designed for. This has been a known limitation for some other mods, but Fluidic Power takes this to a new extreme. However, we still don't know the impact of these calculations _relative_ to the fluid calculations. Does the collecting of statistics really take that much more time than the fluid calculations themselves?
```